src/chain/chain.rs

```diff
@@ -1173,15 +1178,45 @@ impl Chain {
         key_info: SectionKeyInfo,
         proofs: AccumulatingProof,
     ) -> Result<SectionProofBlock, RoutingError> {
-        Ok(SectionProofBlock::new(
-            key_info,
-            self.combine_signatures(proofs)
-                .ok_or(RoutingError::InvalidNewSectionInfo)?,
-        ))
+        let signature = self
+            .check_and_combine_signatures(&key_info, proofs)
+            .ok_or(RoutingError::InvalidNewSectionInfo)?;
+        Ok(SectionProofBlock::new(key_info, signature))
     }
 
-    pub fn combine_signatures(&self, proofs: AccumulatingProof) -> Option<BlsSignature> {
-        proofs.combine_signatures(self.our_info(), self.our_section_bls_keys())
+    pub fn check_and_combine_signatures<S: Serialize + Debug>(
+        &self,
+        signed_payload: &S,
+        proofs: AccumulatingProof,
+    ) -> Option<BlsSignature> {
+        let signed_bytes = serialise(signed_payload)
+            .map_err(|err| {
+                log_or_panic!(
+                    LogLevel::Error,
+                    "{} Failed to serialise accumulated event: {:?} for {:?}",
```

Comment:
Should this be something like "Failed to serialise payload", since the function is generic and there is no requirement the payload actually is `AccumulatedEvent`?

Reply:
While the payload is somewhat generic, it is only for accumulated event.
This is re-enforced by the other parameter being an AccumulatingProof. The assumption that it is not ok to not be able to combine signature is because it is an accumulated event. So I don't think it make sense to update the log statement.

what do you think?

Reply:
Shouldn't then the function take only `AccumulatedEvent`?

Reply:
We have already destructured. Also, it is not the AccumulatedEvent but the RelocateDetails or ElderInfo that we serialize to sign

Reply:
OK. I mean it feels a bit wrong, but it's a very minor issue, so I'll accept it.